### PR TITLE
Fix actors with no footprint not being copied and pasted

### LINKED ISSUE #15896: Map Editor - Some actors aren't copy pasted
Two actors do not copy/paste in the Map Editor : player spawns, and flares.

### PR DESCRIPTION
**Pull request by abcdefg30**
Fixes #15896.

### CODE REVIEW

File: OpenRA.Mods.Common/Traits/World/EditorActorLayer.cs [hunk 1]
```diff
@@ -127,6 +127,21 @@ public EditorActorPreview Add(string id, ActorReference reference, bool initialS
 				list.Add(preview);
 			}
 
+			// Fallback to the actor's CenterPosition for the ActorMap if it has no Footprint
+			if (!preview.Footprint.Any())
+			{
+				var location = worldRenderer.World.Map.CellContaining(preview.CenterPosition);
+
+				List<EditorActorPreview> list;
```
Comment (teinarss) [inline OpenRA.Mods.Common/Traits/World/EditorActorLayer.cs]: Could consider to extract the following section to remove the duplication.

**Review by teinarss — CHANGES_REQUESTED** [review]

File: OpenRA.Mods.Common/Traits/World/EditorActorLayer.cs (hunk 1 above)
Comment (abcdefg30) [reply OpenRA.Mods.Common/Traits/World/EditorActorLayer.cs]: I don't think it is really necessary, but done.

**Comment by teinarss** [comment]
I had to find something to complain about! After all those {} fixes I have
needed to do! :p

**Review by teinarss — APPROVED** [review]

**Review by RoosterDragon — APPROVED** [review]

**Comment by RoosterDragon** [comment]
[Changelog](https://github.com/OpenRA/OpenRA/wiki/Changelog-(bleed)/_compare/886123d133c2bf0d47414e650c7ec5af2f7fe46a)

**Pull request merged**